src/array_serde.rs

```diff
@@ -175,7 +175,6 @@ impl<A, Di, S> serde::de::Visitor for ArrayVisitor<S,Di>
         };
 
         if let Ok(array) = ArrayBase::from_shape_vec(dim, data) {
-            try!(visitor.end());
```

Comment:
Since the docs say "This signals to the MapVisitor that the Visitor does not expect any more items.", it seems strange to call end before making the missing field errors?